### ISSUE TITLE
Literal[negative int]

### ISSUE THREAD

**Issue by crusaderky** [python/mypy#7844]
Python 3.7.4
typing_extensions 3.7.4
mypy git tip ( 0.750+dev.78f2df9d46198563106dd32cb4494678e97d71eb):

mypy returns false positives for negative integer literals.
errors inlined for convenience:
```python
from typing_extensions import Literal


a: Literal[-2] = -2  # error: Incompatible types in assignment (expression has type "int", variable has type "Literal[-2]")
b: Literal[-1] = -1  # error: Incompatible types in assignment (expression has type "int", variable has type "Literal[-1]")
c: Literal[0] = 0
d: Literal[1] = 1
e: Literal[2] = 2
```

**Comment by JukkaL** [comment]
Apparently nobody thought of negative integer literals! This clearly looks like a bug. Either negative literals should be rejected in literal types or they should work as expected.

cc @Michael0x2a

**Comment by Michael0x2a** [comment]
By coincidence, I also ran into this the other day -- the problem is expressions like `-1` are actually `(1).__neg__()` and so we end up using the return type of just `__neg__()`, which is just int.

The easiest fix is probably to special-case the `__neg__` method for ints using a plugin. E.g. intercept `Instance` objects and copy over the `last_known_value` field to the return type.

(If somebody wants to try taking this, feel free. Otherwise, I'll probably work on this later next week.)

**Comment by TH3CHARLie** [comment]
@Michael0x2a I'd like to take this but I may need more info about it, i.e. the very first file I need to look to.

**Comment by Michael0x2a** [comment]
@TH3CHARLie -- sure. 

The change you're trying to make is to return a new callback in the `get_method_hooks` method here: https://github.com/python/mypy/blob/master/mypy/plugins/default.py#L52.

I recommend skimming through some of the existing method hook callbacks to get a feel for how they work, but on a high level what you'll want to do is to modify the return type if the underlying object is:

1. An [`Instance` object](https://github.com/python/mypy/blob/master/mypy/types.py#L690) where the `last_known_value` field is set to an int [LiteralType object](https://github.com/python/mypy/blob/master/mypy/types.py#L1538).
2. An int LiteralType object.

It doesn't matter too much where you add test cases for this, but probably https://github.com/python/mypy/blob/master/test-data/unit/check-literal.test is a reasonable home for them. The [Tests README](https://github.com/python/mypy/blob/master/test-data/unit/README.md) has more info on writing/running test cases.

**Comment by TH3CHARLie** [comment]
@Michael0x2a I have started a PR for this issue. I have finished the callback function but the test keeps failing and  it produces output shown below
```
# test code 
[case testNegativeIntLiteral]
from typing_extensions import Literal

a: Literal[-2] = -2  # E: Incompatible types in assignment (expression has type "int", variable has type "Literal[-2]")
b: Literal[-1] = -1  # E: Incompatible types in assignment (expression has type "int", variable has type "Literal[-1]")
c: Literal[0] = 0
d: Literal[1] = 1
e: Literal[2] = 2
[out]

# output
Expected:
  main:3: error: Incompatible types in assignment (expression has type "int", variable has type "Literal[-2]") (diff)
  main:4: error: Incompatible types in assignment (expression has type "int", variable has type "Literal[-1]") (diff)
Actual:
  main:3: error: Unsupported operand type for unary - ("int") (diff)
  main:4: error: Unsupported operand type for unary - ("int") (diff)
```
However,  invoking mypy directly from the command line does produce errors like @crusaderky 's code did. So I decided to only commit my test for now and maybe we should investigate this inconsistency first.
